Solution

isikach opened the pull request:
![image](https://github.com/mate-academy/py-taxi-service-class-based-views/assets/55293120/8f4698fc-9f0d-4fd0-82c7-16480f68ef85)

isikach committed:
Solution

isikach commented:
![image](https://github.com/mate-academy/py-taxi-service-class-based-views/assets/55293120/9bf05521-61a7-45b7-9787-a68ecdf5ec8a)
![image](https://github.com/mate-academy/py-taxi-service-class-based-views/assets/55293120/d813ec97-62f2-45a5-9fa0-d4846a16c605)
![image](https://github.com/mate-academy/py-taxi-service-class-based-views/assets/55293120/b5814e52-955c-425a-b564-095ec9c233f0)

Serhii-Leonenko commented on taxi/urls.py:
```diff
@@ -1,9 +1,22 @@
+    path("drivers/", DriverListView.as_view(), name="driver-list"),
+    path("drivers/<int:pk>/", DriverDetailView.as_view(),
+         name="driver-detail"),
 ]
```
fix indentation

Serhii-Leonenko commented on taxi/views.py:
```diff
@@ -13,3 +14,32 @@ def index(request):
+
+class ManufacturerListView(generic.ListView):
+    model = Manufacturer
+    queryset = Manufacturer.objects.all()
```
queryset is a redundant attribute, cause you don't customize it

Serhii-Leonenko commented on taxi/views.py:
```diff
@@ -13,3 +14,32 @@ def index(request):
+
+class CarListView(generic.ListView):
+    model = Car
+    queryset = Car.objects.all().select_related("manufacturer")
```
all is redundant attribute

Serhii-Leonenko commented on taxi/views.py:
```diff
@@ -13,3 +14,32 @@ def index(request):
+    model = Driver
+    queryset = Driver.objects.all()
+    template_name = "taxi/driver_list.html"
+    context_object_name = "driver_list"
```
redundant attributes

Serhii-Leonenko commented on taxi/views.py:
```diff
@@ -13,3 +14,32 @@ def index(request):
+
+class DriverDetailView(generic.DetailView):
+    model = Driver
+    queryset = Driver.objects.all().prefetch_related("cars")
```
the same about all

Serhii-Leonenko commented on templates/includes/pagination.html:
```diff
@@ -0,0 +1,11 @@
+      <li class="page-item"><a class="page-link" href="?page={{ page_obj.next_page_number }}">Next</a></li>
+    {% endif %}
+  </ul>
+{% endif %}
```
don't forget to add a blank line to the end of file

Serhii-Leonenko commented on templates/includes/sidebar.html:
```diff
@@ -1,6 +1,18 @@
+<ul class="sidebar-nav list-group">
+  <li class="list-group-item d-flex justify-content-between align-items-center">
+    <a href="{% url "taxi:index" %}" class="flex-grow-1">Home page</a>
+    <span class="badge badge-primary badge-pill">{{ num_home }}</span>
```
what is num_hom?

Serhii-Leonenko commented on templates/taxi/car_detail.html:
definitely you have N + 1 problem here with drivers

Serhii-Leonenko commented on templates/taxi/car_list.html:
```diff
@@ -0,0 +1,22 @@
+  <tbody>
+    {% for car in car_list %}
+      <tr>
+        <th scope="row"><a href={{ car.id }}>{{ car.id }}</a></th>
```
does this link work?

isikach replied:
![image](https://github.com/mate-academy/py-taxi-service-class-based-views/assets/55293120/c2d2a05f-d9d3-457d-b0b6-e4bf904c5895)
![image](https://github.com/mate-academy/py-taxi-service-class-based-views/assets/55293120/1d9ae206-0a69-4234-993e-671cb251bebc)

for sure😎

isikach committed:
1)fixed indentation in urls
2)deleted all redundant attribute
3)added a blank line to the end of file
4)fixed N + 1 problem with drivers
5)* check works link + deleted redundant in html

isikach committed:
1)fixed indentation in urls
2)deleted all redundant attribute
3)added a blank line to the end of file
4)fixed N + 1 problem with drivers
5)* check works link + deleted redundant in html

isikach committed:
1)fixed indentation in urls
2)deleted all redundant attribute
3)added a blank line to the end of file
4)fixed N + 1 problem with drivers
5)* check works link + deleted redundant in html

isikach committed:
1)fixed indentation in urls
2)deleted all redundant attribute
3)added a blank line to the end of file
4)fixed N + 1 problem with drivers
5)* check works link + deleted redundant in html

isikach committed:
1)fixed indentation in urls
2)deleted all redundant attribute
3)added a blank line to the end of file
4)fixed N + 1 problem with drivers
5)* check works link + deleted redundant in html

Serhii-Leonenko commented on taxi/views.py:
```diff
@@ -34,12 +32,11 @@ class CarDetailView(generic.DetailView):
     model = Driver
-    queryset = Driver.objects.all()
     template_name = "taxi/driver_list.html"
     context_object_name = "driver_list"
```
do you really need template_name and context_object_name attributes?

isikach committed:
Update views.py

Deleted unnecessary attributes
